src/test/java/com/faforever/api/clan/ClanControllerTest.java

```diff
@@ -0,0 +1,284 @@
+import com.faforever.api.user.UserRepository;
+import org.codehaus.jackson.JsonNode;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.security.authentication.encoding.ShaPasswordEncoder;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.ResultActions;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.util.Base64Utils;
+import org.springframework.web.context.WebApplicationContext;
+
+import javax.inject.Inject;
+import javax.servlet.Filter;
+import java.util.Collections;
+
+import static org.junit.Assert.assertEquals;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class ClanControllerTest {
+  private static final String OAUTH_CLIENT_ID = "1234";
+  private static final String OAUTH_SECRET = "secret";
+  private MockMvc mvc;
+  private WebApplicationContext context;
+  private Filter springSecurityFilterChain;
+  private ClanRepository clanRepository;
+  private UserRepository userRepository;
+  private ClanMembershipRepository clanMembershipRepository;
+  private PlayerRepository playerRepository;
+  private OAuthClientRepository oAuthClientRepository;
+  private ObjectMapper objectMapper;
+  private ShaPasswordEncoder shaPasswordEncoder;
+  private Player me;
+
+  public ClanControllerTest() {
+    objectMapper = new ObjectMapper();
+    shaPasswordEncoder = new ShaPasswordEncoder(256);
+  }
+
+  @Inject
+  public void init(WebApplicationContext context,
+                   ClanRepository clanRepository,
+                   UserRepository userRepository,
+                   PlayerRepository playerRepository,
+                   OAuthClientRepository oAuthClientRepository,
+                   Filter springSecurityFilterChain,
+                   ClanMembershipRepository clanMembershipRepository) {
+    this.context = context;
+    this.clanRepository = clanRepository;
+    this.userRepository = userRepository;
+    this.playerRepository = playerRepository;
+    this.oAuthClientRepository = oAuthClientRepository;
+    this.springSecurityFilterChain = springSecurityFilterChain;
+    this.clanMembershipRepository = clanMembershipRepository;
+  }
+
+  @Before
+  public void setUp() {
+    mvc = MockMvcBuilders
+        .webAppContextSetup(context)
+        .addFilter(springSecurityFilterChain)
+        .build();
+    me = null;
```

Comment:
A new instance of the test class is instantiated before every `@Test`, which means `me` is always `null` when `setUp()` is called, which means this line is redundant.